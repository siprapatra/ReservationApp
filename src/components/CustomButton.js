import React from "react";
import { Text, TouchableOpacity, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const CustomButton = ({ text, onButtonClick }) => (
  <TouchableOpacity
    style={styles.itemContainer}
    onPress={() => onButtonClick()}
  >
    <Text style={styles.itemText}>{text}</Text>
  </TouchableOpacity>
);

CustomButton.propTypes = {
  text: PropTypes.string,
  onButtonClick: PropTypes.func
};

CustomButton.defaultProps = {
  text: "",
  onButtonClick: () => {}
};

const styles = StyleSheet.create({
  itemContainer: {
    marginBottom: 20,
    marginTop: 10,
    marginHorizontal: 10,
    borderColor: "black",
    borderRadius: 5,
    borderWidth: 1,
    height: 50,
    backgroundColor: "blue",
    justifyContent: "center",
    alignItems: "center"
  },
  itemText: {
    fontSize: 17,
    color: "white"
  }
});

export default CustomButton;
