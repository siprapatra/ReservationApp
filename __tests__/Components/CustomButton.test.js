import React from "react";
import renderer from "react-test-renderer";
import CustomButton from "../../src/components/CustomButton";

describe("Test for Custom Button", () => {
  test("should render the button", () => {
    const snap = renderer
      .create(
        <CustomButton
          text="My Button"
          onItonButtonClickemClick={() => {
            console.log("button clicked");
          }}
        />
      )
      .toJSON();
    expect(snap).toMatchSnapshot();
  });
  test("should render the button without props", () => {
    const snap = renderer.create(<CustomButton />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
