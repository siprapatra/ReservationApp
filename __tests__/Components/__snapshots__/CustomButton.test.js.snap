// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Custom Button should render the button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 1,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 20,
      "marginHorizontal": 10,
      "marginTop": 10,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 17,
      }
    }
  >
    My Button
  </Text>
</View>
`;

exports[`Test for Custom Button should render the button without props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 1,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 20,
      "marginHorizontal": 10,
      "marginTop": 10,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 17,
      }
    }
  >
    
  </Text>
</View>
`;
